### ISSUE TITLE
API Login endpoint always says password is wrong

### ISSUE THREAD

**Issue by jwvanderbeck** [KerbalStuff/KerbalStuff#109]
I'm trying to setup a script for automated releases, but the login endpoint isn't working.  I've triple checked my username and password.  They work fine on the website, but not when sent to the API.

My password does contain symbols, that might be breaking things in the API?  Unfortunately there doesn't appear to be any way on the website to change my password, so I can't test a simpler one.

**Comment by SirCmpwn** [comment]
Did you URL encode your password?

**Comment by jwvanderbeck** [comment]
Its a form field, do you encode those?  In any case I just tried it and got the same error.

**Comment by SirCmpwn** [comment]
Encode it as `multipart/form-data`, and you need to keep cookies around.

**Comment by jwvanderbeck** [comment]
Ok sorry now you are confusing me.  First you mention URLEncode, then you mention multipart/form-data, these are two different things and AFAIK the latter would be no encoding at all on a password.

**Comment by SirCmpwn** [comment]
Yeah, my bad. If you're just copy+pasting curl commands make sure you're taking care of cookies.

**Comment by jwvanderbeck** [comment]
But it doesn't get to that stage.  The API returns an error, so the cookies are irrelevant no?

`{"reason": "Username or password is incorrect", "error": true}`

**Comment by SirCmpwn** [comment]
Ah. I'm not sure, then. I'll test it myself later today.

**Comment by jwvanderbeck** [comment]
Ok the password reset email finally came, and I changed my password to simple text only, and now it works.  So I would say your code isn't handling passwords with symbols in it?  Specifically my old password contained the symbols '@' and '$'

**Comment by SirCmpwn** [comment]
I doubt it. Probably something wrong with the transport, I would guess that '$' is causing problems with bash or whatever.
